flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBResourceContainer.java

```diff
@@ -143,6 +151,12 @@ public ColumnFamilyOptions getColumnOptions() {
                         "We currently only support BlockBasedTableConfig When bounding total memory.");
                 blockBasedTableConfig = (BlockBasedTableConfig) tableFormatConfig;
             }
+            if (rocksResources.isUsingPartitionedIndexFilters()) {
+                blockBasedTableConfig.setIndexType(IndexType.kTwoLevelIndexSearch);
+                blockBasedTableConfig.setPartitionFilters(true);
+                blockBasedTableConfig.setPinTopLevelIndexAndFilter(true);
+                this.overwriteFilterIfExist(blockBasedTableConfig);
+            }
```

Comment:
```suggestion
            if (rocksResources.isUsingPartitionedIndexFilters() && overwriteFilterIfExist(blockBasedTableConfig)) {
                blockBasedTableConfig.setIndexType(IndexType.kTwoLevelIndexSearch);
                blockBasedTableConfig.setPartitionFilters(true);
                blockBasedTableConfig.setPinTopLevelIndexAndFilter(true);
            }
```
This suggestion indicates that full filters is a necessity to enable partitioned index filters, and if the `overwriteFilterIfExist` is changed to always do the overwriting, then it will always return true and the code could be adjusted accordingly.